### DISCUSSION TITLE
Porquê você começou a programar?

### DISCUSSION THREAD

**Discussion by rubenmarcus** [frontendbr/forum#913] [Discussão]
Qual foi a primeira motivação e inspiração sua?E qual seria seu maior sonho/ realização na área?

1 - Comecei por causa do meu pai que hoje é programador ASP.NET, mas começou nos anos 80 com outras linguagens, com 12 anos fazia 'sites' no Dreamweaver e hospedava em sites como VirtualAve e Hypermart. Ai descobri uns scripts do tipo CGI/Perl pra criar um sistema de noticias, ai evolui pro PHP , PHP Nuke, depois WordPress, e aí foi.. dos meus 16 aos 22 eu meio que sofri um hiato com programação deixei de lado total, até voltar a trabalhar na empresa de um conhecido e ai começar de vez com front-end e webdesign. 
Eu vacilei por uns 3 anos fiquei só fazendo freelances e perdi o bonde da evolução JS, mas to tirando o atraso de um ano pra cá, e tenho gostado muito.

2- Fazer um app que realmente mude a vida das pessoas, e ter condição um dia de dedicar 100% do meu tempo para open source.

**Comment by luizbills** [comment]
1 - Quando eu tinha 15 ou 16 anos, o jogo DotA (o clássico) fazia muito sucesso (e ainda deve fazer). Um dia descobri que o dota era feito usando uma engine do jogo Warcraft 3. Lá fui eu meter a cara tentando criar um jogo que fizesse sucesso como o dota. Mas não obtive sucesso com jogos e sim fazendo bibliotecas e auxiliando a comunidade. Naquela época eu nem sabia o que era open source e já estava me auto iniciando :laughing: 

Voltando... a engine do warcraft 3 tinha uma GUI que ajudava a criar o jogo (alguns costumam chamar de programar com blocos). Só que para fazer coisas mais complexas e com mais facilidade você tinha que **programar**. Foi ai que conheci uma linguagem da engine, chamada *JASS (Just Another Scripting Syntax)*. Mal eu sabia que eu me dedicaria tanto e até chegaria a fazer **fama** entre a comunidade nacional (que era bem pequena). Enfim, esse foi o motivo que me levou a aprender programação.

---

2 - Já tive o "sonho" de ser desenvolvedor de jogos. Mas hoje em dia meu foco é manter meu sustento, melhorar minhas habilidades e tornar o meu serviço cada vez mais valioso e prazeroso (trabalho com manutenção de lojas virtuais e desenvovilmento de plugins para WordPress).

**Comment by AlexsandroSA** [comment]
1. Criar jogos (acabei até criando alguns) :bowtie:
2. Criar algo que mude a vida das pessoas de uma forma positiva!

**Comment by leandroruel** [comment]
começei a programar porque odiava o meu antigo emprego, mas precisava dele, ai pra sair dessa dependência, cursei uma faculdade, e meti as caras estudando, dias sem sair aos finais de semana, até tentar uma vaga na área.

**Comment by danielschmitz** [comment]
Comecei a programar em 1995, com BASIC. Fazia curso técnico de eletrotécnica, e tinha o trabalho final de curso que era calcular todas as peças necessárias para se construir um motor. Lembro que eram dezenas de fórmulas, e foi um mês só pro professor explicar cada uma delas. No trabalho final passei tudo isso para o computador e o professor ficou doido, pq ele nunca tinha visto nada parecido (e  passou a usar o programa como gabarito KK). 

Depois dessa experiência fui trabalhar com Pascal, o livro de algoritmo e etrutura de dados era minha bíblia.

**Comment by MatheusWTF** [comment]
Eu comecei em 2013 (sou um zigoto na área praticamente) graças a uma bolsa de estudos que tinha ganhado, aprendi HTML, CSS, e PHP (odiava na época, não uso PHP até hoje por causa do curso kkk). Mas acabei largando porque achava chato.
Ano passado eu resolvi que seria Web Designer, e depois de muito estudar eu acabei dando de cara com JS e me apaixonei, comecei a estudar e hoje não largo mais, estou até fazendo faculdade na área, porém ainda não tenho emprego, nem nunca trabalhei com projeto algum

**Comment by leandroruel** [comment]
@MatheusWTF tem que fazer uns pet projects e subir no github mesmo

**Comment by luizbills** [comment]
@danielschmitz programador roots

**Comment by willycamargo** [comment]
Fiz um "cursinho profissionalizante" daqueles que não ensinam muita coisa. Mas foi suficiente para saber da existência de algumas tecnologias como HTML, CSS e PHP. Além de ter bastante contato com pacote Adobe: Fireworks, Photoshop, Flash e DreamWeaver.

Devido a curiosidade e um pouco de dedicação eu me destaquei diante da turma e consegui um estágio remunerado na mesma escola para auxiliar os professores e até mesmo para ministrar algumas aulas. Com a necessidade de ministrar aulas eu me afundei em estudar e comecei a aprender de verdade 😄

**Comment by phsantiago** [comment]
Comecei por volta da minha 8ª série. Jogava muito Habbo na época e tinha muita curiosidade para saber como funcionava, comecei abrir uns servidores piratas que me deu uma iniciação boa em infra, PHP, HTML, CSS, e Js. Mal eu sabia mas já tava começando no que faço de melhor hoje.

No ensino médio fiz um curso técnico de 1 ano e meio na ETEC, me deu uma base melhor de programação, decidi que era isso que queria e terminando o médio já ingressei em Analise e Desenvolvimento de Sistemas na FATEC, já no começo do curso comecei um estágio já na area.

Hoje em carteira um sou front e em freelas sou fullstack, no geral sou viciado em Javascript, testes e programação funcional. Evolui bastante profissionalmente e cada dia percebo que quanto mais eu estudo, menos sei.

---

Edit: esqueci do sonho haha.
Sonho em fazer algo com código que mude o mundo, criar um NodeJS ou até mesmo um produto como Uber, Waze, Spotify, Netflix.

**Comment by AmandaFonseca** [comment]
O AcessaSP (um programa do estado de SP que oferece espaços públicos com computadores para acesso gratuito à internet) estava oferecendo cursos para a comunidade e eu escolhi manutenção de computadores. Porém na época minha irma me incentivou a fazer um curso na área da saúde porque era mais fácil para arrumar emprego e eu fiz técnico em farmácia. Até que o cunhado da minha irmã começou a trabalhar como desenvolvedor e me mostrou um pouco do que ele fazia, como eu não gostava de trabalhar na farmácia resolvi arriscar fiz Análise de sistemas sem nunca ter programado e sem conhecer nada sobre o mercado tipo de paraquedas mesmo. Pensei várias vezes em desistir e sempre me perguntava:? "Porque eu escolhi esse raio desse curso?",  mesmo achando tudo muito difícil e fora do meu mundo eu mandava bem nas matérias e não reprovava, só reprovei em 1 uma e foi porque não consegui entregar o trabalho a tempo e aos poucos fui pegando gosto pela coisa e hoje amo programação<3 Não sei muito bem o que me motivou mas hoje amo o que faço e quero sempre aprender mais e mais.

**Comment by ninetails** [comment]
Comecei programando fóruns de anime em phpBB. Comecei no PHP e migrei pra front na época do jQuery e "passou nos navegadores, tá ok", pois no PHP eu "trazia muito pensamento pra casa", mesmo não estando trabalhando eu ficava pensando em melhorar os sistemas, fazendo uma arquitetura melhor.

Anos depois continuei em front e tenho agora os mesmos problemas que tinha quando saí de backend :D thx transpilers, frameworks, builders, (pre|post)-processadores

Estou um tanto desiludido com problemas pessoais, não tenho um sonho bem definido, mas diria algo como ter reconhecimento, algo como um projeto que eu participar ganhar um prêmio, ou virar referência.

**Comment by lucasvalhos** [comment]
Em 2004 comecei a jogar um jogo chamado Ragnarok, com não tinha PC em casa eu pagava lan house. Depois de um tempo o jogo começou a cobrar pelo acesso e como eu era curto de grana, ouvi de um carinha em uma lan house que tinha um tal de servidor pirata. Como era muito curioso, peguei minha bike e fui para outra lan ver o que era isso. Baixei o servidor (4gb na época que o máximo de conexão que tínhamos na cidade era 512kb, 1MB), e depois de um tempo decidi criar o meu. Aí conheci linux, c (o emulador era em C), MySql e para fazer o site/painel de controle do servidor comecei a mexer com PHP. E web foi a área que gostei mais. Depois de um hiato (2007-2010), comecei a trabalhar profissionalmente em 2011 e estou até hoje.

Agora meu sonho na área é trabalhar no exterior, e estou me preparando pra isso. Mas por causa dos filhos que eu e minha esposa pretendemos ter, e quero dar uma boa educação para eles.

**Comment by ghost** [comment]
começei a programar porque queria criar uma rede social e porque achei legal
